bin/lib/component.js

```diff
@@ -0,0 +1,166 @@
+const fs = require( 'fs' );
+const path = require( 'path' );
+
+const stringify = require( './parse' ).stringify;
+
+const help = `
+    component
+
+      Read a single file component definition from STDIN or from a file via the
+      -i option and compile it into an ES module to output either to STDOUT or to
+      a file via the -o option. Linked stylesheets will be imported into the
+      component from their external files and included alongside any style elements.
+      Any script tags will be concatenated an used as the JS body of the template
+      file. The remaining HTML will be turned into a template and inlined into the
+      script anywhere $TEMPLATE is encountered. Any aggregated css will be injected
+      into the script as a JS string (JSON.stringify-ed) anywhere $CSS is
+      encountered.
+
+      Any additional script tags encountered with type text/ractive or text/html
+      and an id or name attribute will be automatically turned into partials on the
+      main template. <template> tags with an id attribute will also be turned into
+      partials on the main template.
+
+      script and link tags can specify the path to a file to include directly using
+      the src and href attributes, respectively. Paths to remote files should be
+      relative to the source file.
+
+      -i, --input {file} - file to parse, otherwise STDIN
+          This may also be a directory containing a component in index.html.
+      -o, --output {file} - file to write, otherwise STDOUT
+      -x, --nocsp - don't output expression functions
+      -d, --delimiters {open} {close} - set the plain mustache delimiters (default {{ }})
+      -s, --static {open} {close} - set the static mustache delimiters (default [[ ]])
+      -r, --triple {open} {close} - set the triple mustache delimiters (default {{{ }}})
+      -p, --static-triple {open} {close} - set the static triple delimiters (default [[[ ]]])
+`;
+
+function build ( Ractive, string, output, opts, base ) {
```

Comment:
I'm curious why `Ractive` has to be handed over via args. I believe you can easily do the following since Node does seem to cache modules.

```
var Ractive = require('../../ractive');
```